program-runtime/src/compute_budget.rs

```diff
@@ -308,6 +318,30 @@ impl ComputeBudget {
             prioritization_fee: prioritization_fee_details.get_fee(),
         }
     }
+
+    /// Returns cost of the Poseidon hash function for the given number of
+    /// inputs is determined by the following quadratic function:
+    ///
+    /// 61*n^2 + 542
+    ///
+    /// Which aproximates the results of benchmarks of light-posiedon
+    /// library[0]. These results assume 1 CU per 33 ns. Examples:
+    ///
+    /// * 1 input
+    ///   * light-poseidon benchmark: `18,303 / 33 ≈ 555`
+    ///   * function: `61*1^2 + 542 = 603`
+    /// * 2 inputs
+    ///   * light-poseidon benchmark: `25,866 / 33 ≈ 784`
+    ///   * function: `61*2^2 + 542 = 786`
+    /// * 3 inputs
+    ///   * light-poseidon benchmark: `37,549 / 33 ≈ 1,138`
+    ///   * function; `61*3^2 + 542 = 1091`
+    ///
+    /// [0] https://github.com/Lightprotocol/light-poseidon#performance
+    #[allow(clippy::integer_arithmetic)] // There is no chance of overflow.
```

Comment:
Nit: it seems like technically, the function can overflow; it is just that the caller will never invoke it to overflow. What do you think about using safe arithmetic functions here to return an `Option` and then the caller in `syscall/mod.rs` can just unwrap it there? This might be better design especially since the function is public.